### ISSUE TITLE
Now Playing Info

### ISSUE THREAD

**Issue by XLRradio** [netmix/radio-station#194]
Hi All
 Is it possible to include a Now Playing Widget so we could add the current now playing info from the Stream you put in so it can be visible on the website, and have options to include the size of the font etc?

**Comment by XLRradio** [comment]
@majick777 when will the player be up and running too?

**Comment by tonyzeoli** [comment]
Radio Startion already has a playlisting widget. Start your playlist under the playlist admin and assign it to a show. Add songs to playlist generator and they will appear in the widget.

We have plans to add a sidebar audio player to handle your shoutcast or icecast stream in FREE. We are just starting to look at our options and expect that will come sometime in the next 60 days.

We will offer a sticky footer player in the PRO version, which is probably 45 to 60 days out.

**Comment by XLRradio** [comment]
No I don't mean the playlist function where you add your own playlist. I mean pulling the metadata from the stream. so whatever is playing out on a URL it comes up on the now playing widget.

**Comment by XLRradio** [comment]
@tonyzeoli bit premature weren't u? closing this subject before you get an answer????

**Comment by majick777** [comment]
@XLRradio Yes we will be definitely be exploring the option of reading the stream info and pulling the playlist from that. In order to integrate this properly though, we really do need to get the player in place first.

**Comment by majick777** [comment]
Rechecked my notes and this feature was slated for the Pro version.

**Comment by majick777** [comment]
Sorry I didn't clarify this properly at all as I first thought! A further explanation is required, as the three of us were not on the same page, I somehow assumed you were talking about Icecast/Shoutcast streams, but you didn't actually mention those. I was trying to repurpose this ticket for that, but then didn't do it properly. 

For reference, this is why I had this in mind:
https://stackoverflow.com/questions/55225241/icecast-shoutcast-radio-stream-extracting-now-playing-info

We are not looking at reading this data from the audio stream itself by trying to recognize tracks, which is why the thread was closed.

We will be looking at using the Shoutcast / Icecast APIs to read stream metadata so that it can be displayed in the Current Playlist widget. This feature will probably be in the Pro version, however we are also looking at pulling this kind of information from LibreTime and Azuracast via their APIs as Add-ons, so it's also possible the Shoutcast/Icecast stream metadata integration will be in an Add-on as well instead of Pro. Hope that's clearer!
